typespec-extension/src/code-model-builder.ts

```diff
@@ -1696,275 +1715,201 @@ export class CodeModelBuilder {
     );
   }
 
-  private processSchema(type: Type, nameHint: string): Schema {
+  private processSchemaFromSdkType(type: SdkType, nameHint: string): Schema {
     return this.schemaCache.process(type, nameHint) || fail("Unable to process schema.");
   }
 
-  private processSchemaImpl(type: Type, nameHint: string): Schema {
-    switch (type.kind) {
-      case "Intrinsic":
-        if (isUnknownType(type)) {
-          return this.processAnySchema(type, nameHint);
-        } else {
-          throw new Error(`Unrecognized intrinsic type: '${type.name}'.`);
-        }
-
-      case "String":
-        return this.processConstantSchemaForLiteral(type, nameHint);
+  private processSchemaFromSdkTypeImpl(type: SdkType, nameHint: string): Schema {
+    if (isSdkBuiltInKind(type.kind)) {
+      return this.processBuiltInFromSdkType(type as SdkBuiltInType, nameHint);
```

Comment:
`as SdkBuiltInType` is required here, otherwise there will be compile error. I checked the TCGC code and confirmed with Chenjie, `SdkBuildInKind` contains and only contains `SdkBuiltInType`'s `kind`, so `as` is safe here. But we both don't understand why TS compiler can't do type narrow down here. Please let me know if you have any idea.

Reply:
TCGC should narrow the type, e.g.

```ts
function isSdkBuiltInKind(type: SdkType): type is SdkBuiltInType {
  ...
}
```

Reply:
Yeah, I guess so, but in `else` block, the type is already narrowed down. The screenshot is when I hover over line 1702. So seems the compiler can interprete `if` block's type, but can't do type narrow down. 
![image](https://github.com/Azure/autorest.java/assets/87355844/abcef437-08be-4a6a-9952-55baa585c288)

Reply:
I mean TCGC `isSdkBuiltInKind` method should do the `if` and "narrow down" when it is true. This is feature in TypeScript.